### PR TITLE
Security

### PR DESCRIPTION
**Pull request by AmimiHamza**

### CODE REVIEW

File: packages/insea-connect-backend/src/main/java/ma/insea/connect/chat/group/GroupService.java [hunk 1]
```diff
@@ -37,7 +39,10 @@ public Group saveGroup(GroupDTO groupDTO) {
             m.setIsAdmin(false);
             m.setJoiningDate(new java.sql.Date(System.currentTimeMillis()));
```
Comment (Earthyyy) [inline packages/insea-connect-backend/src/main/java/ma/insea/connect/chat/group/GroupService.java:44]: Joining date should be assigned automatically by spring boot

File: packages/insea-connect-backend/src/main/java/ma/insea/connect/chat/group/GroupService.java [hunk 2]
```diff
@@ -37,7 +39,10 @@ public Group saveGroup(GroupDTO groupDTO) {
...
             membershipRepository.save(m);
-        }      
+        }
+        Membership m=membershipRepository.findById(new MembershipKey(group.getCreator(), group.getId())).get();
```
Comment (Earthyyy) [inline packages/insea-connect-backend/src/main/java/ma/insea/connect/chat/group/GroupService.java:48]: I guess, It's possible to not have another request to the database, you can use the groupDTO object to create the membership and save the group at the end

File: packages/insea-connect-backend/src/main/java/ma/insea/connect/chat/group/GroupService.java [hunk 3]
```diff
@@ -49,8 +54,12 @@ public List<Group> findallgroupsofemail(Long myId) {
         return groups;
     }
     public void deleteGroup(Long groupId) {
-        Group group = groupRepository.findById(groupId).get();
-        groupRepository.delete(group);
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        User user = userRepository.findByUsername(authentication.getName()).orElse(null);
+        if (user.getId() == groupRepository.findById(groupId).get().getCreator()){
+            Group group = groupRepository.findById(groupId).get();
```
Comment (Earthyyy) [inline packages/insea-connect-backend/src/main/java/ma/insea/connect/chat/group/GroupService.java]: extract groupRepository.findById(...) to a variable, so you don't fire up the same request to the database

File: packages/insea-connect-backend/src/main/java/ma/insea/connect/chat/group/GroupService.java [hunk 4]
```diff
@@ -68,9 +77,13 @@ public List<User> findUsers(Long groupId) {
...
     }
     public ResponseEntity<String> addGroupMembers(Long groupId, List<Long> users) {
-        //test if the connected user is an admin
-        
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        User connectedUser = userRepository.findByUsername(authentication.getName()).orElse(null);
+        Membership membership = membershipRepository.findByUserIdAndGroupId(connectedUser.getId(), groupId);
         
+        if(membership == null || !membership.getIsAdmin()) {
+            return ResponseEntity.badRequest().body("You are not allowed to add members to this group");
```
Comment (Earthyyy) [inline packages/insea-connect-backend/src/main/java/ma/insea/connect/chat/group/GroupService.java]: the status code in this case should be 401 unauthorized not 400 bad request.
Another point is that Service should not handle Response entities, That's the responsibility of a controller, if you want to raise an exception, you can simply create a custom exception and make it produce a 401 status code, then you can just throw it in the service with a custom message

File: packages/insea-connect-backend/src/main/java/ma/insea/connect/chat/group/GroupService.java (hunk 3 above)
Comment (AmimiHamza) [reply packages/insea-connect-backend/src/main/java/ma/insea/connect/chat/group/GroupService.java]: done

File: packages/insea-connect-backend/src/main/java/ma/insea/connect/chat/group/GroupService.java (hunk 4 above)
Comment (AmimiHamza) [reply packages/insea-connect-backend/src/main/java/ma/insea/connect/chat/group/GroupService.java]: done

**Pull request merged**